pkg/commands/artifact/k8s.go

```diff
@@ -0,0 +1,203 @@
+	if err != nil {
+		return xerrors.Errorf("get k8s artifacts error: %w", err)
+	}
+
+	err = scanK8sImages(ctx, opt, cacheClient, k8sArtifacts)
+	if err != nil {
+		return xerrors.Errorf("scan kubernetes images error: %w", err)
+	}
+
+	err = scanK8sIac(ctx, opt, cacheClient, k8sArtifacts)
+	if err != nil {
+		return xerrors.Errorf("scan kubernetes iac error: %w", err)
+	}
+
+	return nil
+}
+
+func scanK8sImages(ctx *cli.Context, opt Option, cacheClient cache.Cache, artifacts []*artifacts.Artifact) error {
+	// Disable the lock file scanning
+	opt.DisabledAnalyzers = analyzer.TypeLockfiles
+
+	scannerConfig, scannerOptions, err := initScannerConfig(ctx.Context, opt, cacheClient)
+	if err != nil {
+		return xerrors.Errorf("scanner config error: %w", err)
+	}
+
+	reports := make([]reportk8s.KubernetesReport, 0)
+	for _, artifact := range artifacts {
+		for _, image := range artifact.Images {
+			report, err := k8sScan(ctx.Context, image, imageScanner, scannerConfig, scannerOptions)
+
+			// if an image failed to be scanned, we add the report as error and continue to scan other images
+			if err != nil {
+				reports = append(reports, reportk8s.KubernetesReport{
+					Namespace: artifact.Namespace,
+					Kind:      artifact.Kind,
+					Name:      artifact.Name,
+					Image:     image,
+					Results:   report.Results,
+					Error:     err,
+				})
+
+				continue
+			}
+
+			report, err = filter(ctx.Context, opt, report)
+			if err != nil {
+				return xerrors.Errorf("filter error: %w", err)
+			}
+
+			reports = append(reports, reportk8s.KubernetesReport{
+				Namespace: artifact.Namespace,
+				Kind:      artifact.Kind,
+				Name:      artifact.Name,
+				Image:     image,
+				Results:   report.Results,
+			})
+		}
+	}
+
+	reportk8s.PrintImagesReport(reports)
+
+	return nil
+}
+
+func scanK8sIac(ctx *cli.Context, opt Option, cacheClient cache.Cache, artifacts []*artifacts.Artifact) error {
+	// Disable OS and language analyzers
+	opt.DisabledAnalyzers = append(analyzer.TypeOSes, analyzer.TypeLanguages...)
+
+	// Scan only config files
+	opt.VulnType = nil
+	opt.SecurityChecks = []string{types.SecurityCheckConfig}
+
+	// Skip downloading vulnerability DB
+	opt.SkipDBUpdate = true
+
+	scannerConfig, scannerOptions, err := initScannerConfig(ctx.Context, opt, cacheClient)
+
+	tmpdir, err := ioutil.TempDir("", "trivy-iac")
+	if err != nil {
+		return xerrors.Errorf("create tmp folder error: %w", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	reports := make([]reportk8s.KubernetesReport, 0)
+
+	for _, artifact := range artifacts {
+		filename := filepath.Join(tmpdir, fmt.Sprintf("%s-%s-%s.yaml", artifact.Namespace, artifact.Kind, artifact.Name))
+		file, err := os.Create(filename)
+		if err != nil {
+			return xerrors.Errorf("creating tmp file error: %w", err)
+		}
+		defer file.Close()
+
+		err = artifact.WriteToFile(file)
+		if err != nil {
+			return xerrors.Errorf("error writing artifact to file: %w", err)
```

Comment:
Maybe the logic for persisting files on fs should be done in `trivy-kubernetes`, shouldn't trivy accept only root file folder path  and walk it?